test/disabled-options.test.js

```diff
@@ -66,3 +66,20 @@ t.test('should not apply the delay header', async t => {
     t.equal(response.headers[HEADERS.delay], undefined)
   })
 })
+
+t.test('should not apply headers if the headers option is false', async t => {
+  const fastify = Fastify()
+  await fastify.register(slowDownPlugin, {
+    headers: false
+  })
+  t.teardown(() => fastify.close())
+
+  fastify.get('/', async () => 'Hello from fastify-slow-down!')
+
+  await slowDownAPI(DEFAULT_OPTIONS.delayAfter, () => fastify.inject('/'))
+  const response = await fastify.inject('/')
+
+  t.equal(response.headers[HEADERS.limit], undefined)
+  t.equal(response.headers[HEADERS.remaining], undefined)
+  t.equal(response.headers[HEADERS.delay], undefined)
+})
```

Comment:
I would use [t.notHas](https://node-tap.org/docs/api/asserts/#tnothasfound-pattern-message-extra) instead, just for the sake of readability.